libs/librrgraph/src/io/rr_graph_reader.cpp

```diff
@@ -115,3 +134,140 @@ void load_rr_file(RRGraphBuilder* rr_graph_builder,
             read_rr_graph_name);
     }
 }
+
+static RREdgeId process_rr_edge_override(const std::string& line,
+                                         std::vector<float>& overridden_values,
+                                         const RRGraphView& rr_graph) {
+    std::istringstream iss(line);
+    char ch;
+    RREdgeId edge_id;
+
+    if (std::isdigit(line[0])) {
+        // Line starts with an integer
+        int first;
+        iss >> first;
+        edge_id = (RREdgeId)first;
+    } else if (line[0] == '(') {
+        // Line starts with (first, second)
+        int first, second;
+        iss >> ch >> first >> ch >> second >> ch;
+
+        RRNodeId src_node_id = RRNodeId(first);
+        RRNodeId sink_node_id = RRNodeId(second);
+
+        for (RREdgeId outgoing_edge_id : rr_graph.rr_nodes().edge_range(src_node_id)) {
+            if (rr_graph.rr_nodes().edge_sink_node(outgoing_edge_id) == sink_node_id) {
+                edge_id = outgoing_edge_id;
+                break;
+            }
+        }
+
+        if (!edge_id.is_valid()) {
+            VTR_LOG_ERROR("Couldn't find an edge connecting node %d to node %d\n", src_node_id, sink_node_id);
+        }
+
+    } else {
+        VTR_LOG_ERROR("Invalid line format:  %s\n", line.c_str());
+    }
+
+    float value;
+    while (iss >> value) {
+        overridden_values.push_back(value);
+    }
+
+    return edge_id;
+}
+
+struct t_rr_switch_inf_hash {
+    std::size_t operator()(const t_rr_switch_inf& s) const {
+        std::size_t seed = 0;
+
+        // Helper function for hashing
+        auto hash_combine = [&seed](auto&& val) {
+            seed ^= std::hash<std::decay_t<decltype(val)>>{}(val) + 0x9e3779b9 + (seed << 6) + (seed >> 2);
+        };
+
+        // Combine all relevant fields
+        hash_combine(s.R);
+        hash_combine(s.Cin);
+        hash_combine(s.Cout);
+        hash_combine(s.Cinternal);
+        hash_combine(s.Tdel);
+        hash_combine(s.mux_trans_size);
+        hash_combine(s.buf_size);
+        hash_combine(static_cast<int>(s.power_buffer_type));
+        hash_combine(s.power_buffer_size);
+        hash_combine(s.intra_tile);
+        hash_combine(static_cast<int>(s.type()));
+
+        return seed;
+    }
+};
+
+struct t_rr_switch_inf_equal {
```

Comment:
To ensure that any additional fields added to this data structure are properly handled, I suggest overloading the operator within the struct itself.